sdk/python/lib/pulumi/provider/experimental/analyzer.py

```diff
@@ -634,7 +635,13 @@ def py_type_to_property_type(typ: type) -> PropertyType:
 
 
 def is_union(typ: type):
-    return get_origin(typ) == Union
+    # Check for `Union[a, b]`
+    if get_origin(typ) == Union:
+        return True
+    if sys.version_info >= (3, 10):
+        # Check for `a | b`
+        return get_origin(typ) == types.UnionType
+    return False
```

Comment:
Not sure if there's a cost to `get_origin` but I wonder if it'd be neater/cheaper to do something like:

```python
origin = get_origin(typ)
if origin == Union:
  return True
if sys.version_info >= (3, 10) and origin == types.UnionType:
  return True
return False
```

I mean if we're feeling extra cheeky you could make it all one `return origin == Union or (sys.version_info >= (3, 10) and origin == types.UnionType)`, maybe, but not sure that's better.

Reply:
I like slow, dumb, repetitive code 🤷

Reply:
It reminds me of myself.